### PR TITLE
apiserver: avoid resolving 'localhost'

### PR DESCRIPTION
**Pull request by sttts**
Fixes https://github.com/kubernetes/kubernetes/issues/46767.

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/config_selfclient.go [hunk 1]
```diff
@@ -101,8 +101,20 @@ func LoopbackHostPort(bindAddress string) (string, string, error) {
...
-		// compare MaybeDefaultWithSelfSignedCerts which adds "localhost" to the cert as alternateDNS
 		host = "localhost"
+		// Get ip of local interface, but fall back to "localhost".
+		// Note that "localhost" is resolved with the external nameserver first with Go's stdlib.
+		// So if localhost.<yoursearchdomain> resolves, we don't get a 127.0.0.1 as expected.
+		addrs, err := net.InterfaceAddrs()
+		if err == nil {
+			for _, address := range addrs {
+				// check the address type and if it is not a loopback the display it
+				if ipnet, ok := address.(*net.IPNet); ok && ipnet.IP.IsLoopback() {
```
Comment (dims) [inline staging/src/k8s.io/apiserver/pkg/server/config_selfclient.go]: Should it be `ipnet.IP.IsLoopback()` or `!ipnet.IP.IsLoopback()` as the comment implies?

Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/server/config_selfclient.go]: copy&paste error. the comment is wrong.

**Comment by sttts** [comment]
@liggitt added a small unit test. Guessing lgtm still holds.

File: staging/src/k8s.io/apiserver/pkg/server/config_selfclient.go (hunk 1 above)
Comment (dims) [inline staging/src/k8s.io/apiserver/pkg/server/config_selfclient.go]: cool 👍

**Comment by sttts** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Pull request merged**